### DISCUSSION TITLE
getOptions() on chart widget - called but not reflected on the the chart

### DISCUSSION THREAD

**Discussion by nlvrendell** [filamentphp/filament#9918] [Help]
So I used the `getOptions()` of chart widget to set a dynamic option on the chart. I have a function which is triggered by (dispatch()) to update the data of chart.  The data and the content of getOptions() is update/change as I have used logger to trace its value But, the the title on the chart is not updated. 

Code at getOptions. $this->sum is dynamically updated. 
```
protected function getOptions(): array
    {
        $option = [
            'plugins' => [
                'title' => [
                    'display' => true,
                    'text' => 'Total Sent: '.$this->sum,
                    'position' => 'top',
                ],
            ],
        ];

        logger(['training - option' => $option]);

        return $option;
    }
```

**Comment by mgralikowski** [comment]
I am also trying to implement a dynamic configuration in my charts, in my case it is a custom dashboard with charts and widgets + multiple filters displayed in a modal. Indeed getOptions() is called whenever I apply changes in filters, but those changes don't affect a chart at all (configuration is not reloaded). 

@nlvrendell in your case, try to use the built-in method:

```php
public function getHeading(): string|Htmlable|null
{
    return 'Total Sent: '.$this->sum;
}
```
at least the title of a chart can be dynamic.

**Reply by begueradj** [reply]
> 
> @nlvrendell in your case, try to use the built-in method:
> 
> ```
> public function getHeading(): string|Htmlable|null
> {
>     return 'Total Sent: '.$this->sum;
> }
> ```

You can do it like this too:

```php
protected static ?string $heading = 'Custom header';
```
